File: algorithm/Python3_version/leetcode1155.number-of-dice-rolls-with-target-sum.py
class Solution:
    def numRollsToTarget(self, d: int, f: int, target: int) -> int:
        dp = [[0 for i in range(target+1)] for i in range(d+1)]
        dp[0][0] = 1
        for i in range(1,d+1):
            for j in range(0, target):
                if dp[i-1][j]==0:
                    continue
                for k in range(1, f+1):
                    if j+k <= target:
                        dp[i][j+k] = (dp[i][j+k]+dp[i-1][j])%1000000007
        return dp[d][target]

File: algorithm/leetcode1155.number-of-dice-rolls-with-target-sum.cpp
class Solution {
public:
    int numRollsToTarget(int d, int f, int target) {
    	vector<vector<int>> dp(d+1,vector<int>(target+1, 0));
    	dp[0][0] = 1;
    	int i,j,k;
    	for(i = 1; i <= d; ++i)
    	{
    		for(j = 0; j < target; ++j)
    		{
    			if(dp[i-1][j] != 0)
    			{
    				for(k = 1; k <= f; ++k)
    				{
    					if(j+k <= target)
    						dp[i][j+k] = (dp[i][j+k]+dp[i-1][j])%1000000007;
    				}
    			}
    		}
    	}
    	return dp[d][target];
    }
};